# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd
import numpy as np

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.

#Группируем данные в сводной таблице по задолжностям по признаку количества детей:
data_pivot_Presence_of_children = data.pivot_table(index=['children'], values=['debt','days_employed'], aggfunc={'debt': 'sum', 'days_employed':'count'})

# Переименуем столбцы:
data_pivot_Presence_of_children.columns = ['count', 'debt']

# Рассчитаем отношение задолженностей к общему колличесту кредитов:
data_pivot_Presence_of_children['ratio'] = data_pivot_Presence_of_children['debt'] / data_pivot_Presence_of_children['count']

# Отсортируем сводную таблицу:
data_pivot_Presence_of_children.sort_values(by='ratio', ascending=False, inplace=True)

# Отформатируем столбцы в сводной таблице:
data_pivot_Presence_of_children['ratio'] = data_pivot_Presence_of_children['ratio'].apply(lambda x: "{0:.2f}%".format(x*100))

#Выведем результат:
data_pivot_Presence_of_children.head()

count  debt  ratio
children                    
4            41     4  9.76%
2          2052   194  9.45%
1          4808   444  9.23%
3           330    27  8.18%
0         14091  1063  7.54%

<div class="alert alert-success">
<h2> Комментарий ревьюера <a class="tocSkip"> </h2>

<b>Отлично!✅:</b> Молодец, ты верно вывел сводную таблицу!
</div>

**Вывод:** 

2. Граждане с детьми чаще сталкиваются с трудностями в выплате долгов, чем граждане не имеющие детей. Например, если у домохозяйства есть 4 детей, то вероятность не выплатить долг 9.76 %, а если детей нет, то только 7.54%


<div class="alert alert-success">
<h2> Комментарий ревьюера v2<a class="tocSkip"> </h2>

<b>Отлично!✅:</b> +
</div>

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [23]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.

# Построим сводную табоицу по сумме задолженностей и сумме кредитов в разрезе семейного статуса: 
data_pivot_family_status = data.pivot_table(index=['family_status'], values=['debt','days_employed'], aggfunc={'debt': 'sum', 'days_employed':'count'})

# Переименуем столбцы:
data_pivot_family_status.columns = ['count', 'debt']

# Рассчитаем отношение задолженностей к общей сумме кредитов:
data_pivot_family_status['ratio'] = data_pivot_family_status['debt']/data_pivot_family_status['count']

# Отсортируем сводную таблицу:
data_pivot_family_status.sort_values(by='ratio', ascending=False, inplace=True)

# Отформатируем столбцы в сводной таблице:
data_pivot_family_status['ratio'] = data_pivot_family_status['ratio'].apply(lambda x: "{0:.2f}%".format(x*100))

#Выведем результат:
data_pivot_family_status.head()


count  debt  ratio
family_status                            
Не женат / не замужем   2796   273  9.76%
гражданский брак        4134   385  9.31%
женат / замужем        12261   927  7.56%
в разводе               1189    84  7.06%
вдовец / вдова           951    63  6.62%

<div class="alert alert-success">
<h2> Комментарий ревьюера <a class="tocSkip"> </h2>

<b>Отлично!✅:</b> +
</div>

**Вывод:** 4. Вдовы и вдовцы намного реже не платят по счетам, чем не женатые/не замужние граждане. Например у вдов/вдовцов просрочка по кредитам 6.62%, а у не женатых/не замужем - 9.76%.

<div class="alert alert-success">
<h2> Комментарий ревьюера v2<a class="tocSkip"> </h2>

<b>Отлично!✅:</b> +
</div>

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [24]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.

# Напишем функцию, для расчета количества членов домохозяйств:
def family_members_number(row):
    children = row['children']
    family_status_id = row['family_status_id']
    if family_status_id == 0:
        return children + 1
    if family_status_id == 1:
        return children + 2
    else: return 1

# Применим функцию для расчета количества членов домохозяйств:
data['family_members_number'] = data.apply(family_members_number, axis=1)

# Получим новый столбец доход на душу домохозяйства:
data['income_per_capita'] = data['total_income'] / data['family_members_number']

# Расчитаем квартили в доходах на душу:
q4, q5, q6 = data['income_per_capita'].quantile([0.25,0.5,0.75])

# Напишем функцию для категоризации данных, в зависимости от принадлежности к квартилю: 
def inc_per_capita_categories(row):
    if row <= q4:
        return "Низкий доход на душу"
    if row <= q5:
        return "Средний доход на душу"
    if row <= q6:
        return "Доход выше среднего на душу"
    else:                      
        return "Высокий доход на душу"

# применим функцию и получим новые категории, в зависимости от уровня дохода на душу:
data['inc_per_capita_categories'] = data['income_per_capita'].apply(inc_per_capita_categories)

#data.info()

#Группируем данные в сводной таблице по задолжностям по категориям доходов на душу:
data_pivot_inc_per_capita_categories = data.pivot_table(index=['inc_per_capita_categories'], values=['debt','days_employed'], aggfunc={'debt': 'sum', 'days_employed':'count'})
                                                                                                            
# Переименуем столбцы:
data_pivot_inc_per_capita_categories.columns = ['count', 'debt']
                                                        
# Рассчитаем отношение задолженностей к общей сумме кредитов:
data_pivot_inc_per_capita_categories['ratio'] = data_pivot_inc_per_capita_categories['debt']/data_pivot_inc_per_capita_categories['count']

# Отсортируем сводную таблицу:
data_pivot_inc_per_capita_categories.sort_values(by='ratio', ascending=False, inplace=True)

# Отформатируем столбцы в сводной таблице:
data_pivot_inc_per_capita_categories['ratio'] = data_pivot_inc_per_capita_categories['ratio'].apply(lambda x: "{0:.2f}%".format(x*100))

#Выведем результат:
data_pivot_inc_per_capita_categories.head()

count  debt  ratio
inc_per_capita_categories                      
Низкий доход на душу          5333   506  9.49%
Доход выше среднего на душу   5332   418  7.84%
Средний доход на душу         5333   412  7.73%
Высокий доход на душу         5333   396  7.43%

<div class="alert alert-success">
<h2> Комментарий ревьюера <a class="tocSkip"> </h2>

<b>Отлично!✅:</b> +
</div>

**Вывод:** 3. При анализе рисков, лучше анализировать доход на душу, чем доход "в целом". Чем выше доход на душу, тем вероятность не выплат меньше. Например у квартиля с "самым высоким доходом на душу", вероятность столкнутся с трудностями в оплате кредита 7.43%, а у граждан, из квартиля "с самым низким доходом на душу", вероятность столкнуться с трудностями 9.49%


<div class="alert alert-success">
<h2> Комментарий ревьюера v2<a class="tocSkip"> </h2>

<b>Отлично!✅:</b> +
</div>

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [25]:
# Ваш код будет здесь. Вы можете создавать новые ячейки.

#Группируем данные в сводной таблице по задолжностям по целям кредита:
data_pivot_purp_category = data.pivot_table(index=['purpose_category'], values=['debt','days_employed'], aggfunc={'debt': 'sum', 'days_employed':'count'})

# Переименуем столбцы:
data_pivot_purp_category.columns = ['count', 'debt']

# Рассчитаем отношение задолженностей к общей сумме кредитов:
data_pivot_purp_category['ratio'] = data_pivot_purp_category['debt'] / data_pivot_purp_category['count']

# Отсортируем сводную таблицу:
data_pivot_purp_category.sort_values(by='ratio', ascending=False, inplace=True)

# Отформатируем столбцы в сводной таблице:
data_pivot_purp_category['ratio'] = data_pivot_purp_category['ratio'].apply(lambda x: "{0:.2f}%".format(x*100))

#Выведем результат:
data_pivot_purp_category.head()

count  debt  ratio
purpose_category                            
операции с автомобилем     4279   400  9.35%
получение образования      3988   369  9.25%
проведение свадьбы         2313   183  7.91%
операции с недвижимостью  10751   780  7.26%

<div class="alert alert-success">
<h2> Комментарий ревьюера <a class="tocSkip"> </h2>

<b>Отлично!✅:</b> +
</div>

**Вывод:** 
1. С точки зрения просрочек, наиболее безопасно давать деньги в долг на недвижимость, в отличии от автомобилей и образования. Например процент просрочки на операции с недвижимостью 7.26%, в то время как процент просрочки кредитов на авто - 9.35%.


<div class="alert alert-success">
<h2> Комментарий ревьюера v2<a class="tocSkip"> </h2>

<b>Отлично!✅:</b> +
</div>

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

In [26]:
try:
    data_with_null = pd.read_csv('/datasets/data.csv')
except:
    data_with_null = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')


print(len(data_with_null))
data_with_null[data_with_null.isnull().any(axis=1)]

21525


children  days_employed  dob_years education  education_id  \
12            0            NaN         65   среднее             1   
26            0            NaN         41   среднее             1   
29            0            NaN         63   среднее             1   
41            0            NaN         50   среднее             1   
55            0            NaN         54   среднее             1   
...         ...            ...        ...       ...           ...   
21489         2            NaN         47   Среднее             1   
21495         1            NaN         50   среднее             1   
21497         0            NaN         48    ВЫСШЕЕ             0   
21502         1            NaN         42   среднее             1   
21510         2            NaN         28   среднее             1   

               family_status  family_status_id gender  income_type  debt  \
12          гражданский брак                 1      M    пенсионер     0   
26           женат / замужем                 0      M  госслужащий     0   
29     Не женат / не замужем                 4      F    пенсионер     0   
41           женат / замужем                 0      F  госслужащий     0   
55          гражданский брак                 1      F    пенсионер     1   
...                      ...               ...    ...          ...   ...   
21489        женат / замужем                 0      M    компаньон     0   
21495       гражданский брак                 1      F    сотрудник     0   
21497        женат / замужем                 0      F    компаньон     0   
21502        женат / замужем                 0      F    сотрудник     0   
21510        женат / замужем                 0      F    сотрудник     0   

       total_income                           purpose  
12              NaN                   сыграть свадьбу  
26              NaN                       образование  
29              NaN  строительство жилой недвижимости  
41              NaN  сделка с подержанным автомобилем  
55              NaN                   сыграть свадьбу  
...             ...                               ...  
21489           NaN              сделка с автомобилем  
21495           NaN                           свадьба  
21497           NaN        строительство недвижимости  
21502           NaN  строительство жилой недвижимости  
21510           NaN           приобретение автомобиля  

[2174 rows x 12 columns]

*Ответ:* 'Пропуски наблюдаются в 2174 строках из 21525. Это чуть больше 10% всех записей. Пропуски характерны для столбцов с количественными переменными total_income и days_employed. 
Возможно люди неохотно хотят рассказывать о своем доходе и общем трудовом стаже.'

Факторы появления пропусков - доход и количество дней трудового стажа.


<div class="alert alert-success">
<h2> Комментарий ревьюера v2<a class="tocSkip"> </h2>

<b>Отлично!✅:</b> Главную роль все же играют человеческий фактор и технические сбои:)
</div>

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* Заполнять пропуски в количественных переменных нужно наиболее характерными значениями. Доходы подверженны выбросам (экстремально высоким/низким значениям), когда небольшое кол-во наблюдений сильно влияют на среднее значение во всей выборке. Классические аналогии, это когда один человек ест мясо, другой капусту, но в среднем они едят голубцы. Или средняя температура по больнице, или пример с Биллом Гейтсом и баром - когда Бил Гейтс заходит в бар, то все посетители бара в среднем миллиардеры.

<div class="alert alert-success">
<h2> Комментарий ревьюера <a class="tocSkip"> </h2>

<b>Отлично!✅:</b> Совершенно верно!
</div>

### Шаг 4: общий вывод.

Напишите ваш общий вывод.

1. С точки зрения просрочек, наиболее безопасно давать деньги в долг на недвижимость, в отличии от автомобилей и образования. Например процент просрочки на операции с недвижимостью 7.26%, в то время как процент просрочки кредитов на авто - 9.35%.

2. Граждане с детьми чаще сталкиваются с трудностями в выплате долгов, чем граждане не имеющие детей. Например, если у домохозяйства есть 4 детей, то вероятность не выплатить долг 9.76 %, а если детей нет, то только 7.54%

3. При анализе рисков, лучше анализировать доход на душу, чем доход "в целом". Чем выше доход на душу, тем вероятность не выплат меньше. Например у квартиля с "самым высоким доходом на душу", вероятность столкнутся с трудностями в оплате кредита 7.43%, а у граждан, из квартиля "с самым низким доходом на душу", вероятность столкнуться с трудностями 9.49%

4. Вдовы и вдовцы намного реже не платят по счетам, чем не женатые/не замужние граждане. Например у вдов/вдовцов просрочка по кредитам 6.62%, а у не женатых/не замужем - 9.76%.

 <div class="alert alert-success">
<h2> Комментарий ревьюера <a class="tocSkip"> </h2>

    
<b>Отлично!✅:</b> Молодец, верный вывод! Но все это нужно изложить после каждого задания:)
</div>

<div style="border:solid Chocolate 2px; padding: 40px">
Андрей, давай подведем итоги. Ты большой молодец, что старался и проделал такую большую работу. Мне понравилось , что ты без проблем вывел таблицы!
    
    Разберем над чем стоит поработать:

    1. Необходимо выводы написать после каждого задания.
    
    2. Указать два фактора, влияющие на появление пропусков.
    
    Пересмотри работу и исправь ошибки. Обязательно пиши, если я могу тебе еще чем-то помочь. Удачи!)
  

<div class="alert alert-success">
<h2> Комментарий ревьюера v2<a class="tocSkip"> </h2>

<b>Отлично!✅:</b> Андрей, молодец, что так быстро все откорректировал. Работа принята, хорошего вечера:)
</div>